### PR TITLE
Issue 330 more gef editor

### PR DESCRIPTION
**Pull request by theanuradha**
> Replace with style - if dataset styling object is right-clicked, a popup menu should allow the styling to be replaced with another type (plain vs linear styling).

Is yet to be done but will do in separate PR

### CODE REVIEW

**Comment by IanMayo** [comment]
Nice work. Couple of missing features:
- [x] when we drag/drop a chart, it should replace the "target" chart, pushing the drop target down on in the sequence.  In the current implementation the "index" parameter is always -1, and the chart always appears at the bottom of the list of charts. **Update: I've fixed this**
- [x] we can drag/drop a dataset onto an axis landing pad.  When this happens, we need to create an axis, to hold the dataset.  So, you should clone the dataset's current axis, add it to the chart (replacing the landing pad), and then add the dataset to the new axis.
- [x] when we have multiple axes on a chart, we should be able to re-order them via drag/drop

File: info.limpet.stackedcharts.ui/src/info/limpet/stackedcharts/ui/editor/figures/ChartFigure.java [hunk 1]
```diff
@@ -14,28 +18,56 @@
...
+
+  public void setName(final String name)
   {
     chartNameLabel.setText("Chart: " + name);
     // cache font for AxisNameFigure
     if (boldFont == null)
     {
-      FontData fontData = chartNameLabel.getFont().getFontData()[0];
+      final FontData fontData = chartNameLabel.getFont().getFontData()[0];
       boldFont =
```
Comment (IanMayo) [inline info.limpet.stackedcharts.ui/src/info/limpet/stackedcharts/ui/editor/figures/ChartFigure.java:71]: @theanuradha - shouldn't we be using a Font Registry for this?

If we do create one, I guess we could create it in the Activator for this plugin.

**Comment by IanMayo** [comment]
Note to Ian: re-run Debrief build once this is merged:
https://github.com/debrief/debrief/pull/1776

File: info.limpet.stackedcharts.ui/src/info/limpet/stackedcharts/ui/editor/figures/ChartFigure.java (hunk 1 above)
Comment (theanuradha) [inline info.limpet.stackedcharts.ui/src/info/limpet/stackedcharts/ui/editor/figures/ChartFigure.java:71]: right now we statically keep boldFont. will port it to activator base Font Registry

**Comment by theanuradha** [comment]
@IanMayo 

> when we drag/drop a chart, it should replace the "target" chart, pushing the drop target down on in the sequence. In the current implementation the "index" parameter is always -1, and the chart always appears at the bottom of the list of charts. Update: I've fixed this

Fix seems wrong and Drag & Drop stopped working. I have re-fix with what you expect https://github.com/theanuradha/limpet/commit/696a84d5971ec437f53dff104da7b452b89a73f7

**Comment by theanuradha** [comment]
@IanMayo  added support to move dataset to Axis landing pad. 
https://github.com/debrief/limpet/pull/333/commits/4598e82df2fc324e1effff0d5715586d2df9deec

**Comment by theanuradha** [comment]
@IanMayo  

> when we have multiple axes on a chart, we should be able to re-order them via drag/drop

https://github.com/debrief/limpet/pull/333/commits/ad2eceb8db6b9daf0d8fba163ddb75563465b99e

we can merge now

File: info.limpet.stackedcharts.ui/src/info/limpet/stackedcharts/ui/editor/commands/MoveAxisCommand.java [hunk 2]
```diff
@@ -8,33 +8,54 @@
 
 public class MoveAxisCommand extends Command
```
Comment (IanMayo) [inline info.limpet.stackedcharts.ui/src/info/limpet/stackedcharts/ui/editor/commands/MoveAxisCommand.java:9]: @theanuradha - I've played around with the logic here, to tidy the code. It works for me, and I'll do the merge. But, please could you double-check that MoveAxis and MoveChart still work correctly?

**Comment by theanuradha** [comment]
@IanMayo  All looks good !

**Pull request merged**